Runtime/Runner/Scenes/LanguageSelectorController.cs

```diff
@@ -0,0 +1,240 @@
+            {
+                Destroy(child.gameObject);
+            }
+        }
+
+        // Spawn one UI item per selected language
+        foreach (string lang in SimvaPlugin.Instance.SelectedLanguages)
+        {
+            GameObject item = Instantiate(languageItemPrefab, languageGridLayout.transform);
+            foreach(string languageCode in languages.Keys) {
+                if(languages[languageCode] == lang) {
+                    item.name = languageCode;
+                    item.SetActive(true);
+                    continue;
+                }
+            }
+        }
+    }
+
+    void GetLanguages()
+    {
+        languages = new Dictionary<string, string>();
+
+        TextAsset[] filler = Resources.LoadAll<TextAsset>("Localization");
+        foreach (TextAsset obj in filler) {
+            if (obj.name == "lang") {
+                JObject jObject = JObject.Parse(obj.text);
+                var code="";
+                var name="";
+                var custom=false;
+                foreach (var entry in jObject) {
+                    if(entry.Key=="code") {
+                        code=(string)entry.Value;
+                    }
+                    if(entry.Key=="displayName") {
+                        name=(string)entry.Value;
+                    }
+                    if(entry.Key=="custom") {
+                        custom=(bool)entry.Value;
+                    }
+                }
+                var modifName=name + " [" + code + "]";
+                if(!languages.ContainsKey(code)) {
+                    languages.Add(code, modifName);
+                } else {
+                    //if(custom) {
+                    //    languages.Add(code, modifName);
+                    //}
+                }
+            }
+        }
+        Debug.Log("Languages : " + languages.Count);
+    }
+
+    //Creates an initializes Json Files array to be used by myDictionary
+    void SetUpJSONFiles(bool useDefault)
+    {
+        var lang=Language;
+        if (useDefault) {
+            lang = GetLanguageFromTitle(SimvaPlugin.Instance.LanguageByDefault);
+        }
+
+        Object[] filler = Resources.LoadAll("Localization/" + lang + "/" + "Dictionaries", typeof(TextAsset));
+
+        if (filler == null || filler.Length == 0)
+        {
+            Debug.LogError("No JSON Files in Dictionaries directory found (Localization/" + lang + "/" + "Dictionaries) !");
+        }
+        
+        List<TextAsset> json = new List<TextAsset>();
+        foreach (Object file in filler)
+        {
+            json.Add((TextAsset)file);
+        }
+
+#if UNITY_EDITOR
+        foreach (var t in json)
+            Debug.Log("JSON File added for Language " + lang + " : " + t.name);
+#endif
+        if(useDefault) {
+            defaultJsonFiles = json;
+        } else {
+            jsonFiles = json;
+        }
+    }
+
+    //Fills myDictionary with all JSON files
+    void FillDictionary(bool useDefault)
```

Comment:
Similar concern to SetUpJSONFiles, but in this case it should receive the List\<TextAsset\> by parameter, and return the dictionary.

Reply:
fixed